ui/src/app/pages/appAnalysis/apps-list.tsx

```diff
@@ -9,40 +10,108 @@ import { Button } from 'app/components/buttons';
 import { FadeBox } from 'app/components/containers';
 import { FlexColumn, FlexRow } from 'app/components/flex';
 import { ListPageHeader } from 'app/components/headers';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
 import colors from 'app/styles/colors';
-import { withCurrentWorkspace } from 'app/utils';
-import { AnalyticsTracker } from 'app/utils/analytics';
+import { reactStyles, withCurrentWorkspace } from 'app/utils';
+import { APP_LIST } from 'app/utils/constants';
+import { WorkspacePermissionsUtil } from 'app/utils/workspace-permissions';
+
+const styles = reactStyles({
+  fadeBox: {
+    margin: 'auto',
+    marginTop: '1rem',
+    width: '95.7%',
+  },
+  startButton: {
+    paddingLeft: '0.5rem',
+    height: '2rem',
+    backgroundColor: colors.secondary,
+  },
+  appsLabel: {
+    color: colors.primary,
+    fontWeight: 600,
+    fontSize: '14px',
+    lineHeight: '24px',
+    paddingBottom: '0.5rem',
+  },
+});
 
 export const AppsList = withCurrentWorkspace()((props) => {
   const { workspace } = props;
+  const [selectedApp, setSelectedApp] = useState('');
+  const [showSelectAppModal, setShowSelectAppModal] = useState(false);
+
+  const canWrite = (): boolean => {
```

Comment:
It seems like this would execute every time that the page re-renders. Would it make sense to convert this to a state variable?

Reply:
The reason i used constant was because the user's permission level will not change once the page is rendered. And my understanding is that state variables are expected to have a change in values caused by user's action or other reason like result from Api.

Reply:
I agree with Neha: this will be constant, so a state variable is not necessary.

But this also means that we can make this a simple constant instead of a function, for the same reason.

Reply:
Hmmm, I do not think it is a big deal, but I do not think that state variables need to be something that changes (I could be wrong).

I think that: 
`const [canWrite, setCanWrite] = useState(workspace.accessLevel);`
would work. If we are certain that permission level will not change while the user is on the page, we could even do something like:
`const [canWrite, ] = useState(workspace.accessLevel)`
that would prevent creating a set function that will not be used.

Reply:
I do not think that this is a big enough deal to prevent you from merging but something to consider.

Reply:
Just saw your comment @jmthibault79 . I think that makes sense. Thanks to both of you for explaining. I'm good with a constant.